Index: gcc-4.5.2/libgcc/config/lm32/_ashlsi3.S
===================================================================
--- gcc-4.5.2.orig/libgcc/config/lm32/_ashlsi3.S	2011-01-23 12:20:02.000000000 +0100
+++ gcc-4.5.2/libgcc/config/lm32/_ashlsi3.S	2011-01-23 12:20:10.000000000 +0100
@@ -30,6 +30,10 @@
         .global __ashlsi3
         .type __ashlsi3,@function
 
+        .global __ashlsi3_0
+        .hidden __ashlsi3_0
+        .type __ashlsi3_0,@function
+
         .align 4        
 __ashlsi3:
         /* Only use 5 LSBs, as that's all the h/w shifter uses.  */
@@ -111,4 +115,4 @@
 __ashlsi3_0:
         ret
                 
-        
\ No newline at end of file
+        
Index: gcc-4.5.2/libgcc/config/lm32/_ashrsi3.S
===================================================================
--- gcc-4.5.2.orig/libgcc/config/lm32/_ashrsi3.S	2011-01-23 12:19:58.000000000 +0100
+++ gcc-4.5.2/libgcc/config/lm32/_ashrsi3.S	2011-01-23 12:20:07.000000000 +0100
@@ -27,6 +27,10 @@
 
         .global __ashrsi3
         .type __ashrsi3,@function
+
+        .global __ashrsi3_0
+        .hidden __ashrsi3_0
+        .type __ashrsi3_0,@function
                                                   
 __ashrsi3:
         /* Only use 5 LSBs, as that's all the h/w shifter uses.  */
@@ -107,4 +111,4 @@
         sri     r1, r1, 1
 __ashrsi3_0:
         ret
-        
\ No newline at end of file
+        
Index: gcc-4.5.2/libgcc/config/lm32/_lshrsi3.S
===================================================================
--- gcc-4.5.2.orig/libgcc/config/lm32/_lshrsi3.S	2011-01-23 12:19:46.000000000 +0100
+++ gcc-4.5.2/libgcc/config/lm32/_lshrsi3.S	2011-01-23 12:20:05.000000000 +0100
@@ -28,6 +28,10 @@
         .global __lshrsi3
         .type __lshrsi3,@function
 
+        .global __lshrsi3_0
+        .hidden __lshrsi3_0
+        .type __lshrsi3_0,@function
+
 __lshrsi3:
         /* Only use 5 LSBs, as that's all the h/w shifter uses.  */
         andi    r2, r2, 0x1f
